src/common/adxl345.rs

```diff
@@ -0,0 +1,174 @@
+use crate::common::math_utils::Vector3;
+use crate::common::movement_sensor::{MovementSensor, MovementSensorSupportedMethods};
+
+use super::board::{Board, BoardType};
+use super::config::Kind::BoolValue;
+use super::config::{Component, ConfigType};
+use super::i2c::I2CHandle;
+use super::movement_sensor::MovementSensorType;
+use super::registry::ComponentRegistry;
+use super::status::Status;
+
+use std::collections::BTreeMap;
+use std::mem::size_of;
+use std::sync::{Arc, Mutex};
+
+// This module represents an implementation of the MPU-6050 gyroscope/accelerometer
+// as a Movement Sensor component
+
+pub(crate) fn register_models(registry: &mut ComponentRegistry) {
+    if registry
+        .register_movement_sensor("accel-adxl345", &ADXL345::from_config)
+        .is_err()
+    {
+        log::error!("accel-adxl345 type is already registered");
+    }
+}
+
+const READING_START_REGISTER: u8 = 50;
+const STANDBY_MODE_REGISTER: u8 = 45;
+
+pub struct ADXL345 {
+    i2c_handle: I2cHandleType,
+    i2c_address: u8,
+}
+
+impl ADXL345 {
+    pub fn new(mut i2c_handle: I2cHandleType, i2c_address: u8) -> anyhow::Result<Self> {
+        let bytes: [u8; 2] = [STANDBY_MODE_REGISTER, 8];
+        i2c_handle.write_i2c(i2c_address, &bytes)?;
+        Ok(Self {
+            i2c_handle,
+            i2c_address,
+        })
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn from_config(
+        cfg: ConfigType,
+        board: Option<BoardType>,
+    ) -> anyhow::Result<MovementSensorType> {
+        if board.is_none() {
+            return Err(anyhow::anyhow!(
+                "actual board is required to be passed to configure ADXL-345"
+            ));
+        }
+        let board_unwrapped = board.unwrap();
+        match cfg {
+            ConfigType::Static(cfg) => {
+                let i2c_handle: I2cHandleType;
+                if let Ok(i2c_name) = cfg.get_attribute::<&'static str>("i2c_bus") {
+                    i2c_handle = board_unwrapped.get_i2c_by_name(i2c_name.to_string())?;
+                } else {
+                    return Err(anyhow::anyhow!(
+                        "i2c_bus is a required config attribute for ADXL-345"
+                    ));
+                };
+                if match &cfg.attributes {
+                    None => false,
+                    Some(attrs) => match attrs.get("use_alt_i2c_address") {
+                        Some(BoolValue(value)) => *value,
+                        _ => false,
+                    },
+                } {
+                    return Ok(Arc::new(Mutex::new(ADXL345::new(i2c_handle, 29)?)));
+                }
+                Ok(Arc::new(Mutex::new(ADXL345::new(i2c_handle, 83)?)))
+            }
+        }
+    }
+
+    pub fn close(&mut self) -> anyhow::Result<()> {
+        // put the MPU in the sleep state
+        let off_data: [u8; 2] = [STANDBY_MODE_REGISTER, 0];
+        if let Err(err) = self.i2c_handle.write_i2c(self.i2c_address, &off_data) {
```

Comment:
is the sleep command used for other purpose than when closing the ADXL?

Reply:
using the driver in rdk as a reference, there seems to be no other use case